Cpp/odin-views/app.cpp

```diff
@@ -135,17 +147,27 @@ namespace {
             // Create user
             fostlib::pg::connection cnx{fostgres::connection(config, req)};
             auto ref = odin::reference();
-            odin::create_user(cnx, ref, fostlib::coerce<fostlib::string>(fed_response_data["identity"]["id"]));
+            auto const identity_id = fostlib::coerce<fostlib::string>(fed_response_data["identity"]["id"]);
+            odin::create_user(cnx, ref, identity_id);
+
             if ( odin::is_module_enabled(cnx, "opts/full-name") ){
-                odin::set_full_name(cnx, ref,
-                    fostlib::coerce<fostlib::string>(fed_response_data["identity"]["id"]),
+                odin::set_full_name(cnx, ref, identity_id,
                     fostlib::coerce<fostlib::string>(fed_response_data["identity"]["full_name"]));
             }
+
             if ( odin::is_module_enabled(cnx, "opts/email") ){
-                odin::set_email(cnx, ref,
-                    fostlib::coerce<fostlib::string>(fed_response_data["identity"]["id"]),
+                odin::set_email(cnx, ref, identity_id,
                     fostlib::coerce<fostlib::email_address>(fed_response_data["identity"]["email"]));
             }
+
+            // Add membership to group
+            if ( odin::is_module_enabled(cnx, "authz") ){
```

Comment:
I think we need a much better way of doing this, or we're going to trigger SQL each time this is looked at.

Reply:
Agree, do you have any idea?

Reply:
We have to have something that caches this. It could be done in the function itself I guess. Would need to look across the code and see what's what.